datafusion/common/src/tree_node.rs

```diff
@@ -769,6 +770,263 @@ impl<T> Transformed<T> {
     }
 }
 
+/// [`Container`] contains elements that a function can be applied on or mapped. The
+/// elements of the container are siblings so the continuation rules are similar to
+/// [`TreeNodeRecursion::visit_sibling`] / [`Transformed::transform_sibling`].
+pub trait Container<'a, T: 'a>: Sized {
+    fn apply_elements<F: FnMut(&'a T) -> Result<TreeNodeRecursion>>(
+        &'a self,
+        f: F,
+    ) -> Result<TreeNodeRecursion>;
+
+    fn map_elements<F: FnMut(T) -> Result<Transformed<T>>>(
```

Comment:
Could we also add some documentation here about what map_elements is (perhaps just a map back to `TreeNode::map_children`?)

Reply:
Sure, I added comments in https://github.com/apache/datafusion/pull/13467/commits/2097bc7603097f4106d68be9f1b6a281d4bdbb7e to clarify how these methods are used.